jfr-daemon/src/main/java/com/newrelic/jfr/agent/AgentController.java

```diff
@@ -0,0 +1,96 @@
+package com.newrelic.jfr.agent;
+
+import com.newrelic.jfr.daemon.DaemonConfig;
+import com.newrelic.jfr.daemon.JFRUploader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.text.ParseException;
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import javax.management.*;
+import jdk.jfr.Configuration;
+import jdk.jfr.Recording;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AgentController {
+  private static final Logger logger = LoggerFactory.getLogger(AgentController.class);
+
+  private final DaemonConfig config;
+  private final JFRUploader uploader;
+  private Recording recording;
+
+  private volatile boolean shutdown = false;
+
+  private final ExecutorService executorService =
+      Executors.newFixedThreadPool(
+          2,
+          r -> {
+            Thread result = new Thread(r, "NRJFRAgent");
+            result.setDaemon(true);
+            return result;
+          });
+
+  public AgentController(JFRUploader uploader, DaemonConfig config) {
+    this.uploader = uploader;
+    this.config = config;
+  }
+
+  // This needs to be exposed to JMX / k8s
+  public void shutdown() {
+    shutdown = true;
+  }
```

Comment:
This method isn't called at all. Is it relevant when running as a `javaagent`?

Reply:
It's probably not needed for the simple agent, but I would definitely like the ability to dynamically disable JFR (e.g. via a JMX call) when running as an agent extension. What do you think - shall we leave it in or take it out of the simple agent?

Reply:
Making a decision - removing, will add back for the extension.